### PR TITLE
ADAPT-3266: Global Header (Desktop and Mobile)

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- SAA Global Header (both desktop and mobile styles) sans Main Nav
- On the Storyblok side, the masthead schema now allow either a masthead picker (for SAA Homesite masthead) or a global header picker (for all other SAA subsites like Travel/Study). Global header picker is also a single option selector. This has been added to Basic Page, Dark background page, Trip and Trip Filter page.

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Go to https://deploy-preview-185--stanford-alumni-uat.netlify.app/travel-study/
2. Compared global header to Figma design. Check all breakpoints. Check that the when a Hero is added like on this page, it is tucked underneath the global header (absolute positioned) which shows through the header's transparency.
2. Go to https://deploy-preview-185--stanford-alumni-uat.netlify.app/travel-study/curated-collections/
3. Check that for this page (Dark Background Page), the global header is showing solid black color and that it's relative positioned so no content below the global header is tucked underneath it.
3. Go to https://deploy-preview-185--stanford-alumni-uat.netlify.app/travel-study/destinations/
4. Check that for the Destination page, it's the same behavior as https://deploy-preview-185--stanford-alumni-uat.netlify.app/travel-study/   (Hero tucked under header)
3. Go to https://deploy-preview-185--stanford-alumni-uat.netlify.app/ and check that the homesite masthead is the same as before.
4. Look at code

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -1,25 +1,26 @@
 const activeEnv =
-  process.env.GATSBY_ACTIVE_ENV || process.env.NODE_ENV || "development";
+  process.env.GATSBY_ACTIVE_ENV || process.env.NODE_ENV || 'development';
```
Comment (yvonnetangsu) [inline gatsby-config.js:2]: Just double quotes to singles for this file

File: gatsby-config.js [hunk 2]
```diff
@@ -1,25 +1,26 @@
...
-  "localFooterPicker.localFooter",
-  "mastheadPicker.masthead",
-  "perkCard.perkPicker",
-  "perkCardHorizontal.perkPicker",
-  "storyCard.storyPicker",
-  "alertPicker.alert",
-  "verticalNav.verticalNav",
-  "tripCard.trip",
+  'eventCard.eventPicker',
+  'globalHeaderPicker.globalHeader',
```
Comment (yvonnetangsu) [inline gatsby-config.js:15]: Did add this item to storyblokRelations. Now the user can swap out the SAA masthead to any global headers (different Storyblok components)

File: src/components/identity/global-header/GlobalHeaderStyles.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export const root = ({ hasHero, isDark } = {}) =>
+  dcnb(
+    'global-header-desktop su-w-full su-hidden lg:su-block su-border-b su-border-solid su-border-black-90',
+    {
+      'su-bg-saa-black': !hasHero && isDark,
+      'su-bg-gradient-to-b su-from-masthead-black-top su-to-masthead-black-bottom su-backface-hidden':
+        hasHero || !isDark,
+      'su-absolute': hasHero,
+    }
+  );
```
Comment (yvonnetangsu) [inline src/components/identity/global-header/GlobalHeaderStyles.js:12]: Add different classes depending on whether the page has a Hero banner and whether it's a Dark Background Page.

File: src/components/identity/masthead.js [hunk 4]
```diff
@@ -26,11 +27,8 @@ const Masthead = ({ blok: { mainNav, utilityNav }, blok, hasHero, isDark }) => {
         width="full"
         className="masthead-mobile su-relative lg:su-hidden su-bg-cardinal-red-xdark"
       >
-        <nav
-          aria-label="Utility Menu"
-          className="su-w-full su-text-right su-bg-digital-red"
-        >
-          <ul className="su-cc su-list-unstyled su-inline-block su-link-white su-link-no-underline su-link-regular su-text-14 sm:su-text-16 children:su-inline-block children:su-ml-[2em] children:su-mb-0 children:su-leading-[4rem] su-underline-offset">
+        <nav aria-label="Utility Menu" className={styles.utilNavMobile}>
+          <ul className={styles.utilNavMenuMobile}>
```
Comment (yvonnetangsu) [inline src/components/identity/masthead.js:31]: Some global header styles are shared with the old homesite masthead so might as well clean this up a bit.

File: src/components/search/openSearchModalButton.js [hunk 5]
```diff
@@ -1,30 +1,19 @@
...
-        className="su-inline-block su-w-[2.2rem] lg:su-w-20 lg:su-ml-6"
-      />
-    </Button>
-  </>
+  <button
+    type="button"
+    className="search-button su-flex-shrink-0 su-w-40 su-h-40 su-ml-auto su-items-center su-justify-content-center lg:su-w-auto lg:su-h-auto lg:su-pl-16 lg:su-pr-13 lg:su-pt-2 su-rounded-full lg:su-text-18 su-text-white su-leading-none su-border-2 su-border-solid su-border-digital-red-xlight su-transition-colors hocus:su-bg-cardinal-red-xxdark xl:hocus:su-bg-cardinal-red-xdark lg:hocus:su-no-underline"
+    aria-label="Search Stanford Alumni sites"
+    onClick={(e) => setModalOpen(true)}
+  >
```
Comment (yvonnetangsu) [inline src/components/search/openSearchModalButton.js:10]: No huge changes here. There's really no need to use the Decanter React button as it provides no benefit and it's causing a Proptype warning, so I'm replacing with a regular html button.

File: src/types/storyblok/SBLinkType.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import PropTypes from 'prop-types';
+
+// This is the Storyblok Link schema
+export const SBLinkType = PropTypes.shape({
+  id: PropTypes.string,
+  linktype: PropTypes.string,
+  url: PropTypes.string,
+  fieldtype: PropTypes.string,
+  cached_url: PropTypes.string,
+});
```
Comment (yvonnetangsu) [inline src/types/storyblok/SBLinkType.js:10]: Custom Proptype for the Storyblok Link component

File: src/types/NavItemType.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import PropTypes from 'prop-types';
+import { SBLinkType } from './storyblok/SBLinkType';
+
+export const NavItemType = {
+  text: PropTypes.string,
+  link: SBLinkType,
+};
```
Comment (yvonnetangsu) [inline src/types/NavItemType.js:7]: I'll do more to hook this one up in the main nav PR later.

**Review by mattanglin — APPROVED** [review]
lgtm 👍🏼

**Comment by mattanglin** [comment]
@yvonnetangsu Out of curiosity, why the different naming conventions of masthead vs header? And as we expand our Storyblok/Gatsby instance to include more microsites do you see the additional sites just having variations of the globalHeader?

**Comment by yvonnetangsu** [comment]
> @yvonnetangsu Out of curiosity, why the different naming conventions of masthead vs header? And as we expand our Storyblok/Gatsby instance to include more microsites do you see the additional sites just having variations of the globalHeader?

Hi @mattanglin! Good questions. The naming wasn't very deliberate. I just use global header since that's what it's been called in tickets 😄  I believe the intention is that most if not all the microsites will use this same header. I can see abstracting the styling out a bit further later on if there turns out there will be variations.

**Comment by yvonnetangsu** [comment]
Happy to change the name also if we decide to later. I can see it being name "local masthead" or "saa masthead" etc 😂

**Comment by mattanglin** [comment]
> Happy to change the name also if we decide to later. I can see it being name "local masthead" or "saa masthead" etc 😂

Cool! globalHeader totally works! I was just curious as I'm still learning the naming conventions we have here. Things like "masthead" and "lockup" are new terms for me! 🤓

**Comment by yvonnetangsu** [comment]
> > Happy to change the name also if we decide to later. I can see it being name "local masthead" or "saa masthead" etc 😂
> 
> Cool! globalHeader totally works! I was just curious as I'm still learning the naming conventions we have here. Things like "masthead" and "lockup" are new terms for me! 🤓

Admittedly naming here isn't always super consistent 😄  I think we were trying to get rid of the term "lockup" at some point because no one outside of SWS/UComm know what that means, but looks like it's somehow haunting us still.

**Pull request merged**